packages/story-editor/src/components/panels/design/pageAttachment/pageAttachment.js

```diff
@@ -131,6 +131,11 @@ function PageAttachmentPanel() {
   const { getProxiedUrl, checkResourceAccess } = useCORSProxy();
 
   const populateUrlData = useDebouncedCallback(async (value) => {
+    // Nothing to fetch for tel: or mailto: links.
+    if (!value.startsWith('http://') || !value.startsWith('https://')) {
```

Comment:
Would it be better to just check for `tel:` and `mailto:` here instead and return only if the value starts with those?

Even though currently, only `withProtocol` URLs are used with `populateUrlData` then still seems more correct -- for `populateUrlData` not assuming it's used correctly. Thoughts?

Reply:
We could add an `isValidURL` check here if we're worried about the URL being incorrect.

Opted for checking `http`/`https` in case we want to support more protocols in the future. So an _allowlist_ rather than a _denylist_.

Reply:
Ok, it's just that if there's a URL that doesn't start with `http(s)://` and also not with `tel:` nor `mailto:` that ends up here,  it would change the behaviour. So I'd prefer a denylist here :) But if you think it's ok then let's keep it, it's unlikely to happen.

Reply:
> Ok, it's just that if there's a URL that doesn't start with `http(s)://` and also not with `tel:` nor `mailto:` that ends up here, it would change the behaviour.

How so? We cannot populate URL data for anything that isn't an http or https link. Hence returning early here.

Reply:
> How so? We cannot populate URL data for anything that isn't an http or https link. Hence returning early here.

No, we can't, but we're setting URL as invalid and remove the icon if that's the case. Now we would just return without doing it.

Reply:
Maybe I'm not following you here 🤔 

`handleChangeUrl` only calls `populateUrlData` for valid (aka legit-looking) URLs, and then if `populateUrlData` can't actually reach it, it's marked as invalid again.

The behavior I'm expecting:

1. Type `foo:bar` into field -> marked as invalid
2. Type `blahblah` into field -> gets turned into `https://blahblah` -> looks valid -> try to fetch data -> marked as invalid
3. Type `example.com` into field -> gets turned into `https://example.com` -> looks valid -> try to fetch data -> valid
4. Type `mailto:foobar` into field -> is valid

Note:

I just realized the `||` in this condition needs to be an `&&`. Fixing that right now.

Reply:
> Maybe I'm not following you here 

The thought was that _if_ an URL ends up inside `populateUrlData` that doesn't have the protocol nor `tel` nor `mailto` _then_ the behaviour would change (because then we would just return instead of setting the URL invalid + removing the icon). Meaning that currently, `populateUrlData` assumes that the url that gets to it is already validated. Which is ok since currently it's true -- `populateUrlData` is only called with valid URLs. The idea was mainly that I'd prefer not assuming that a function is always called with valid URLs but if you think it's OK here, we can keep it as is.

Reply:
I'd prefer making this assumption and having the validation in one place instead of multiple places.